playbooks/compute/nto/roles/configurecluster/tasks/detect_sno.yml

```diff
@@ -1,21 +1,18 @@
----
-- name: Retrieve worker nodes
+- name: Get Nodes information
   kubernetes.core.k8s_info:
+    api_version: v1
     kind: Node
-    label_selectors:
-      - node-role.kubernetes.io/worker
-  register: worker_nodes_info
+  register: nodes_info
 
-- name: Retrieve master nodes (for SNO case)
-  kubernetes.core.k8s_info:
-    kind: Node
-    label_selectors:
-      - node-role.kubernetes.io/master
-  register: master_nodes_info
+- name: Register nodes names
+  ansible.builtin.set_fact:
+    nodes_names: "{{ nodes_info.resources | map(attribute='metadata.name') | list }}"
 
-- name: Check if this is SNO (single node with master+worker roles)
+- name: Verify if it's SNO cluster
```

Comment:
in this PR you are using 2 diff methods of verifying in it'sSNO, Isuggest sticking to one. 

best is -   

```
`    - name: Get Nodes information
      kubernetes.core.k8s_info:
        api_version: v1
        kind: Node
      register: nodes_info

    - name: Verify if it's SNO cluster
      ansible.builtin.set_fact:
        sno_cluster: true
      when: nodes_info.resources | length == 1`
```